api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,16 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						// if 'name' is a file then make sure that only path is trimmed out and not the file name itself 
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
You're right, in openmrs-core this function is always called with `name` parameter as filename, but at the same time it is always called with `keepFullPath == true`, so the case causing an error never occured.